client/client.go

```diff
@@ -2,40 +2,30 @@ package client
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"io"
 
-	"github.com/EventStore/EventStore-Client-Go/stream_position"
+	"errors"
 
-	"github.com/EventStore/EventStore-Client-Go/connection"
-	"github.com/EventStore/EventStore-Client-Go/persistent"
+	"github.com/EventStore/EventStore-Client-Go/internal/protoutils"
 	persistentProto "github.com/EventStore/EventStore-Client-Go/protos/persistent"
+	"github.com/EventStore/EventStore-Client-Go/types"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
 
-	"github.com/EventStore/EventStore-Client-Go/client/filtering"
-	"github.com/EventStore/EventStore-Client-Go/direction"
-	"github.com/EventStore/EventStore-Client-Go/errors"
-	"github.com/EventStore/EventStore-Client-Go/internal/protoutils"
-	"github.com/EventStore/EventStore-Client-Go/messages"
 	api "github.com/EventStore/EventStore-Client-Go/protos/streams"
-	stream_revision "github.com/EventStore/EventStore-Client-Go/streamrevision"
 )
 
-type Configuration = connection.Configuration
-
-func ParseConnectionString(str string) (*connection.Configuration, error) {
-	return connection.ParseConnectionString(str)
-}
-
 // Client ...
 type Client struct {
-	grpcClient connection.GrpcClient
-	Config     *connection.Configuration
+	grpcClient *grpcClient
+	Config     *Configuration
 }
 
 // NewClient ...
-func NewClient(configuration *connection.Configuration) (*Client, error) {
-	grpcClient := connection.NewGrpcClient(*configuration)
+func NewClient(configuration *Configuration) (*Client, error) {
+	grpcClient := NewGrpcClient(*configuration)
```

Comment:
```suggestion
	client := NewGrpcClient(*configuration)
```
`grpcClient` is already a type in this context (`type grpcClient struct`), so better to rename the variable to avoid confusions